### ISSUE TITLE
Opening a conversation often doesn't display all toots 

### ISSUE THREAD

**Issue by deutrino** [stom79/mastalab#390]
Since the last few updates, when I tap a toot in a fairly large conversation that's ongoing, not all toots in the conversation is displayed. I know this because I can see other toots in the given conversation in my timeline, which didn't appear when I first opened it. When I tap one of those, I get a different subset of toots in the conversation. 

It looks like this is maybe an attempt at threading? Or maybe just a regression?

As a user, when I tap any toot in a conversation, I want to be able to access *all* toots in that conversation. This become particularly important when it's a large conversation spread out over several days, where I'm unlikely to see other subthreads in my timeline. 

If threading is the goal, maybe collapsing subthreads would be a better UI?

It's difficult to provide an example since perceiving this bug requires noticing toots you didn't previously see, in a timeline. 

---

Instance: I've seen this on multiple instances 

Mastalab version: 1.8.5 

Android version: 7.1.2

**Comment by stom79** [comment]
It's not a regression but an enhancement. Before, the whole conversation was opened, even replies not related to each others. This behaviour made more difficult to understand long conversations. Now the app works like the web app. But, you can still expand to the whole conversation by clicking in the expand button at the to right.
I guess you would like an option allowing to expand the whole conversation by default ;)

**Comment by deutrino** [comment]
Okay, if it's an enhancement I would say there isn't nearly enough of a cue that some toots are hidden. Also, the button is just an anonymous button that doesn't visually give any indication what it does or that it should be used.

If I could have anything I wanted ;) I would love to see toots collapsed directly in the conversation view, like:

  * toot that starts the conversation 
  * [collapsed subthread with 12 toots]
  * [collapsed subthread with 4 toots]
  * toot that started the subthread I tapped on
  * reply in that subthread 
  * reply in that subthread which I tapped on to open the conversation view
  * reply in that subthread 
  * reply in that subthread 
  * [collapsed subthread with 15 toots] 
  * [collapsed subthread with 8 toots] 

So to sum up, the thing that bothers me about the enhancement - which I think is a great idea actually! - is that it's not currently obvious that I'm only seeing a subset of the conversation.

**Comment by deutrino** [comment]
Sorry, I was typing on my phone when you replied. I will use the button of course. I await your thoughts on the other concerns :)

**Comment by stom79** [comment]
In fact, the app doesn't decide the order of these toots and uses the context: https://github.com/tootsuite/documentation/blob/master/Using-the-API/API.md#context

When you click on a toot, it opens the conversation. So, before this "clicked" toot are displayed the ancestors and after are displayed the descendants.

For having the whole conversation whatever the toot clicked in that conversation, the app gather the first toot, ie: ancestors->get(0) and then gets the context of this first toot to retrieve all its descendants. That's what is done when clicking the expand button.
